### ISSUE TITLE
Using ws+unix with a specific path

### ISSUE THREAD

**Issue by apocas** [websockets/ws#236]
How can I use socketPath and simultaneously specifying an endpoint path?

Use case:
socketPath: /var/run/docker.sock
path: /container/71501a8ab0f8/attach/ws

If i do: ws+unix:///var/run/docker.sock/container/71501a8ab0f8/attach/ws it will not work because url.parse in WebSocket.js#L581 will return the full path after ws+unix://

**Comment by fearenales** [comment]
Only for further reference, a workaround for this issue is:

```js
var url = require('url');
var WebSocket = require('ws');

var address  = url.parse('ws+unix:///var/run/docker.sock');
address.path = '/container/71501a8ab0f8/attach/ws';
var ws       = new WebSocket(address);
```

This works because WebSocket.js doesn't re-parse the url if it already is in a parsed object.

Tks @nuxlli

**Comment by coolaj86** [comment]
request.js handles it like this:

```
http://unix:/absolute/path/to/unix.socket:/request/path
```

What do you think about a pull request that allows that format?

**Comment by alandoherty** [comment]
WebSocket still seems to try and parse it as a URL even if you `url.parse` beforehand, was this ever implemented?

**Comment by lpinca** [comment]
Refs: #528

**Comment by lpinca** [comment]
Fixed by 060b275d978ff7d095a92863fc3a1f7bcea878ed, closing.
